src/openassetio-python/package/openassetio/pluginSystem/PythonPluginSystemManagerImplementationFactory.py

```diff
@@ -37,48 +37,97 @@ class PythonPluginSystemManagerImplementationFactory(ManagerImplementationFactor
-    based plugins. It uses the platform-native delimiter.  Searched left
-    to right.
+    directories to search for @ref
+    openassetio.pluginSystem.PythonPluginSystemManagerPlugin based
+    plugins. It uses the platform-native delimiter.  Searched left to
+    right. Plugins found here will take precedence over those discovered
+    through package entry points. Note: this search path is entirely
+    independent of **PYTHONPATH** and doesn't not need to be a subset of
+    those paths. This allows OpenAssetIO plugins to be managed entirely
+    independent of the python runtime if desired, and masked from
+    `import` statements.
+
+    @envvar **OPENASSETIO_DISABLE_ENTRYPOINTS_PLUGINS** when set,
+    disables entry point based plugin discovery. This can be useful if
+    it is not in use, to avoid unnecessary filesystem access during
+    library initialization. **OPENASSETIO_PLUGIN_PATH** plugins take
+    precedence over any entry point based ones.
     """
 
     ## The Environment Variable to read the plug-in search path from
     kPluginEnvVar = "OPENASSETIO_PLUGIN_PATH"
+    ## The Environment Variable to control the discovery of entry point based plugins
+    kDisableEntryPointsEnvVar = "OPENASSETIO_DISABLE_ENTRYPOINTS_PLUGINS"
+
+    ## The name of the ManagerPlugin entry point for entry point
+    ## discovered plugins.
+    kPackageEntryPointName = "openassetio.manager_plugin"
+
+    def __init__(self, logger, paths=None, disableEntryPointsPlugins=None):
+        """
+        Creates a new factory. The factory scans for plugins lazily on
+        the first invocation of @ref identifiers or @ref instantiate.
 
-    def __init__(self, logger, paths=None):
+        @param logger @fqref{log.LoggerInterface} "LoggerInterface"
+        implementation that will be used to output information about
+        plugin loading.
+
+        @param paths `str` Paths to search for plugins. Defaults to the
+        value of the @ref kPluginEnvVar environment variable.
+
+        @param disableEntryPointsPlugins `bool` Controls whether
+        package entry point based plugin discovery is allowed. Defaults
+        to False unless the @ref kDisableEntryPointsEnvVar environment
+        variable is set.
+        """
 
         super(PythonPluginSystemManagerImplementationFactory, self).__init__(logger)
 
         self.__pluginManager = None
         self.__paths = paths
+        self.__disableEntryPointsPlugins = disableEntryPointsPlugins
 
     def __scan(self):
         """
         Scans for PythonPluginSystemManagerPlugins, and registers them
         with the factory instance.
-
-        @param paths `str` A searchpath string to search for plug-ins.
-        If None, then the contents of the Environment Variable
-        @ref kPluginEnvVar is used instead.
         """
 
-        if not self.__paths:
+        if self.__disableEntryPointsPlugins is None:
```

Comment:
[Consider][Minor] Took me a second to parse, maybe a quick "if disable entry points member not set, attempt to use environment variable, otherwise false" comment?

Reply:
[Consider] Could you do this in `init`, making `self.__disableEntryPointsPlugins` an effective invariant?

Reply:
Yeah, I think we did it that way originally to allow the env to be changed after construction, but before scanning. Given that the paths/etc.. are constructor params, I think you're right, and its better there. Changed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/768/commits/4a5fc9e5ce493567c8b771f6104e9701f19ea3c3).